### ISSUE TITLE
Add a `--licensepath` / `EASYCONFIG_LICENSEPATH` option

### ISSUE THREAD

**Issue by besserox** [hpcugent/easybuild-framework#1711]
Many easyconfig files use something like to find their licence file.
```
license_file = os.path.join(os.getenv('HOME'), "licenses", "intel", "license.lic")
```
The path is hardcoded to `$HOME/licenses`.

I believe it would be useful to have a `--licensepath` / `EASYCONFIG_LICENSEPATH` option.
This way, the user can easily customize the location of their license files.

This would also make it more consistent with other paths (sourcepath, buildpath, installpath, etc.)

What do you think?

**Comment by boegel** [comment]
@besserox makes sense to me, avoid hardcoding and adding flexibility is a plus, just as long we make the default value `$HOME/licenses` for backward compatibility.

We can also add a corresponding constant like `LICENSEPATH` that can be used in easyconfig files, similar to like we did for `$HOME` (cfr. #1607).
